### ISSUE TITLE
When generating a bundle (dev/prod) clean up generated frontend files if none existed before.

### ISSUE THREAD

**Issue by caalador** [vaadin/flow#16783]
If the framework generates a dev/prod bundle and there has been no frontend files in the project.
All generated files should be automatically removed
- package.json
- package-lock.json
- types.d.ts
- tsconfig.json
- vite.config.ts
- vite.generated.ts
- node_modules/
- .pnpmfile.cjs
- .npmrc

This makes it easier for the developer to see changed and know what should be committed to the code repository

**Comment by lgf10** [comment]
We have commited these files to our repositories a long time ago, mainly because a missing package-lock.json would significantly increase build time and various documentation recommended it, e.g. https://vaadin.com/docs/v14/flow/upgrading/v10-13#6-build-and-maintain-the-v14-project
A missing package-lock.json no longer increases the build time noticably, so we have removed these files now as keeping them updated was somewhat of a chore.
It would be nice if this was documented somewhere, that it is no longer recommended to keep these files in version control if no customizations have been made. I have had to check the maven plugin source code to find this issue and why the files were being deleted if they did not exist when running the maven goal.
